packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -228,11 +228,11 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         this._attachState = this.containerRuntime.attachState !== AttachState.Detached && this.existing ?
             this.containerRuntime.attachState : AttachState.Detached;
 
-        this.bindToContext = () => {
+        this.bindToContext = async () => {
             assert(this.bindState === BindState.NotBound, 0x13b /* "datastore context is already in bound state" */);
             this.bindState = BindState.Binding;
             assert(this.channel !== undefined, 0x13c /* "undefined channel on datastore context" */);
-            bindChannel(this.channel);
+            await bindChannel(this.channel);
```

Comment:
Similar to other places, it's extremely hard to say if conversion to async here would not open pandora box of bugs.
Given that this code is used in synchronous workflows (i.e., storing a handle as a value in a map - that will cause attachment of data store / DDS that are backed by such handle, which previously was fully synchronous process), it's hard to believe that async parts of this process can happen later, when user has a chance to continue to modify objects that we are trying to snapshot. Similar as in other places, we need to ensure we capture enough state (revisions) synchronously to represent state of universe before allowing async processes run off that revision.

Reply:
I don't understand the full implications of this. Does it mean that it will be very hard to get things right? Or we should enforce that some things remain synchronous?